src/app/editor/code/editor-container/codegen/vue/vue.service.ts

```diff
@@ -1,105 +1,59 @@
 import { Injectable } from '@angular/core';
-import { CodeGenFacade, XlayersNgxEditorModel } from '../codegen.service';
-import { SharedCodegen, Template } from '../shared-codegen.service';
+import { WebBlocGenService } from '@xlayers/web-blocgen';
 
 @Injectable({
   providedIn: 'root'
 })
-export class VueCodeGenService implements CodeGenFacade {
-
-  constructor(private sharedCodegen: SharedCodegen) {}
+export class VueCodeGenService {
+  constructor(private webBlocGen: WebBlocGenService) {}
 
   buttons() {
     return {};
   }
 
-  generate(ast: SketchMSLayer): Array<XlayersNgxEditorModel> {
-    return [{
-      uri: 'README.md',
-      value: this.generateReadme(),
-      language: 'text/plain',
-      kind: 'text'
-    }, {
-      uri: 'Xlayers.vue',
-      value: this.generateComponent(ast),
-      language: 'html',
-      kind: 'vue'
-    }, {
-      uri: 'xlayers.test.ts',
-      value: this.generateComponentSpec(),
-      language: 'javascript',
-      kind: 'vue'
-    }];
+  generate(data: SketchMSData) {
+    return [
+      {
+        kind: 'text',
+        value: this.renderReadme(data.meta.app),
+        language: 'markdown',
+        uri: `README.md`
+      },
+      ...(data.pages as any).flatMap(page => {
```

Comment:
Can we use a type instead of `any`?

Reply:
I had a type issue because flatMap and ECMAScript stage stuffs, seem to be fixed since.